### PR TITLE
Apply environment variable both for LocalBranch and RemoteBranch

### PR DESCRIPTION
**Pull request by qinezh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Common/Git/GitUtility.cs [hunk 1]
```diff
@@ -154,49 +139,66 @@ private static GitRepoInfo GetRepoInfoCore(string directory)
...
+                    remoteBranch = remoteBranch.Substring(index + 1);
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogInfo($"For git repo <{repoRootPath}>, can't find remote branch in this repo and fallback to use local branch [{localBranch}]: {ex.Message}");
+                remoteBranch = localBranch;
             }
-            return RunGitCommandAndGetFirstLine(repoRootPath, GetLocalBranchCommand);
+            return new Tuple<string, string>(localBranch, remoteBranch);
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/Git/GitUtility.cs]: use `Tuple.Create`

File: src/Microsoft.DocAsCode.Common/Git/GitUtility.cs [hunk 2]
```diff
@@ -154,49 +139,66 @@ private static GitRepoInfo GetRepoInfoCore(string directory)
...
         }
 
         // Many build systems use a "detached head", which means that the normal git commands
         // to get branch names do not work. Thankfully, they set an environment variable.
-        private static string GetLocalBranchNameFromDetachedHead(string repoRootPath)
+        private static Tuple<string, string> GetBranchNameFromDetachedHead(string repoRootPath)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Common/Git/GitUtility.cs]: >Tuple<string, string> [](start = 23, length = 21)

Keep return type, and create Tuple in GetBranchNames()?

**Comment by vwxyzh** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**